src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -127,39 +125,48 @@ void ExecutorWorker::execute() {
             bool bypassProgressComplete = false;
             switch (syncOp->type()) {
                 case OperationType::Create: {
-                    handleCreateOp(syncOp, job, hasError, ignored);
+                    executorExitInfo = handleCreateOp(syncOp, job, ignored);
                     break;
                 }
                 case OperationType::Edit: {
-                    handleEditOp(syncOp, job, hasError, ignored);
+                    executorExitInfo = handleEditOp(syncOp, job, ignored);
                     break;
                 }
                 case OperationType::Move: {
-                    handleMoveOp(syncOp, hasError, ignored, bypassProgressComplete);
+                    executorExitInfo = handleMoveOp(syncOp, ignored, bypassProgressComplete);
                     break;
                 }
                 case OperationType::Delete: {
-                    handleDeleteOp(syncOp, hasError, ignored, bypassProgressComplete);
+                    executorExitInfo = handleDeleteOp(syncOp, ignored, bypassProgressComplete);
                     break;
                 }
                 default: {
                     LOGW_SYNCPAL_WARN(_logger, L"Unknown operation type: "
                                                        << syncOp->type() << L" on file "
                                                        << SyncName2WStr(syncOp->affectedNode()->name()).c_str());
-                    _executorExitCode = ExitCode::DataError;
-                    _executorExitCause = ExitCause::Unknown;
-                    hasError = true;
+                    executorExitInfo = ExitCode::DataError;
                 }
             }
 
-            if (hasError) {
+            // If an operation fails but is correctly handled by handleExecutorError, execution can proceed.
+            if (executorExitInfo.cause() == ExitCause::OperationCanceled) {
+                setProgressComplete(syncOp, SyncFileStatus::Error);
```

Comment:
setProgressComplete should be called only if !bypassProgressComplete.

Reply:
Done here: [e09c87c](https://github.com/Infomaniak/desktop-kDrive/pull/356/commits/e09c87c48a98f3671f335d9531dc0cb4bd42592b)